src/Transport/Receiving/MessagePump.cs

```diff
@@ -243,13 +243,18 @@ async Task ProcessMessage(Task<Message> receiveTask)
                         await receiver.SafeAbandonAsync(pushSettings.RequiredTransactionMode, lockToken).ConfigureAwait(false);
                     }
                 }
-                catch (Exception onErrorException)
+                catch (Exception onErrorException) when (ExceptionShouldTriggerCriticalError(onErrorException))
```

Comment:
what's going to happen with the message in case this catch block (which calls `SafeAbandonAsync`) isn't executed?